roles/custom/matrix-synapse-reverse-proxy-companion/templates/labels.j2

```diff
@@ -119,6 +119,44 @@ traefik.http.routers.matrix-synapse-reverse-proxy-companion-public-client-synaps
 ############################################################
 {% endif %}
 
+{% if matrix_synapse_reverse_proxy_companion_container_labels_internal_client_synapse_admin_api_enabled %}
```

Comment:
It seems like `matrix_synapse_reverse_proxy_companion_container_labels_internal_client_synapse_admin_api_enabled` actually enables 2 things right now: `/_synapse/client` and `/_synapse/admin`.

For the "public" exposure of these, we have dedicated variables: `matrix_synapse_reverse_proxy_companion_container_labels_public_client_synapse_client_api_enabled` and `matrix_synapse_reverse_proxy_companion_container_labels_public_client_synapse_admin_api_enabled`.

So it's odd to have a single variable (`matrix_synapse_reverse_proxy_companion_container_labels_internal_client_synapse_admin_api_enabled`) exposing both right now.

I suggest to create an additional variable (`matrix_synapse_reverse_proxy_companion_container_labels_internal_client_synapse_client_api_enabled`) and use that one to decide whether to expose `/_synapse/client`.

Of course, you can wire the new variable (`matrix_synapse_reverse_proxy_companion_container_labels_internal_client_synapse_client_api_enabled`) in `group_vars/matrix_servers` the same way (based on `matrix_bot_draupnir_room_hijack_enabled`).

I don't know if Draupnir room hijacking requires both `/_synapse/admin` and `/_synapse/client` though. If it only requires the former, then there's no need to enable the latter based on `matrix_bot_draupnir_room_hijack_enabled`.

-----------

You've added labels to `matrix-synapse-reverse-proxy-companion`, which handles the workers-being-enabled case.

Similar labels should be added to `roles/custom/matrix-synapse/templates/labels.j2` as well.
Along with those, variables for each of these should be defined in the `matrix-synapse` role. Example:

- `matrix_synapse_container_labels_internal_client_synapse_admin_api_enabled`
- `matrix_synapse_container_labels_internal_client_synapse_admin_api_traefik_rule`
- `matrix_synapse_container_labels_internal_client_synapse_admin_api_entrypoints`
- `matrix_synapse_container_labels_internal_client_synapse_client_api_enabled`
- `matrix_synapse_container_labels_internal_client_synapse_client_api_traefik_rule`
- `matrix_synapse_container_labels_internal_client_synapse_client_api_entrypoints`

The `matrix-synapse-reverse-proxy-companion` role tends to inherit (via `group_vars/matrix_overrides`) various variable values from `matrix-synapse` ([here](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/e000cbf0f42874bec05611d57a2922259da292bc/group_vars/matrix_servers#L4473-L4474) is an example), so you can wire the new `matrix-synapse-reverse-proxy-companion` variables (enabled, entrypoints, etc.) the same way.

Reply:
I should note that when it comes to this PR i am blindly taking the labels from the issue in question. 

So like it makes sense that these flaws exist. And i need to thank you for this very thuroughly written review. 

I will try to take a look at this during the week to see how much of this stuff i can fix on my own based on all this valuable feedback and trying to look at other roles more closely to understand how they are plumbed. (Primarily Synapse admin i think i will need to look closer at as i think it helps a lot.)

As for the exact API surface Draupnir needs for its admin commands i will come back with an answer on that front as i honestly dont know but i know i can find out.

Reply:
That's understandable! Thank you for looking into it!